examples/chatbot/prepare_chatbot_data.py

```diff
@@ -0,0 +1,156 @@
+# Copyright 2019 The Forte Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import argparse
+import os
+import importlib
+import logging
+
+import torch
+import texar.torch as tx
+
+import data_utils
+
+logging.basicConfig(level=logging.INFO)
+
+parser = argparse.ArgumentParser()
+parser.add_argument("--file-name", default="data/source/dataset.txt",
+                    help="Data directory to read the files from")
+parser.add_argument("--output-dir", default="data/",
+                    help="Output directory to write the pickled files")
+parser.add_argument("--config-data", default="config_data",
+                    help="Configuration file")
+args = parser.parse_args()
+
+device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+
+config_data = importlib.import_module(args.config_data)
+
+
+def main():
+    train_data = []
+    eval_data = []
+    test_data = []
+
```

Comment:
Many codes for preparing the data could be boilerplates, let's consider how to standardize them and allow users to implement this with a few lines of code.

Reply:
This file is essentially doing the following

- Split the dataset into train, eval and test
- Augment the data with contextual information
- Generate Negative examples
- Saving it as RecordData

The functions for (1)-(3) are found in `data_utils.py`. We may impose some standard on the structure. But I am not sure if we want to move it as part of Forte core.